src/GafferScene/UDIMQuery.cpp

```diff
@@ -0,0 +1,315 @@
+
+
+using namespace IECore;
+using namespace IECoreScene;
+using namespace Gaffer;
+using namespace GafferScene;
+
+IE_CORE_DEFINERUNTIMETYPED( UDIMQuery );
+
+size_t UDIMQuery::g_firstPlugIndex = 0;
+
+UDIMQuery::UDIMQuery( const std::string &name )
+	:	ComputeNode( name )
+{
+	storeIndexOfNextChild( g_firstPlugIndex );
+	addChild( new ScenePlug( "scene", Plug::In ) );
+	addChild( new StringPlug( "uvName", Plug::In, "uv" ) );
+	addChild( new StringPlug( "extraAttributes", Plug::In, "" ) );
+	addChild( new FilterPlug( "filter" ) );
+	addChild( new CompoundObjectPlug( "udimInformation", Plug::Out, new IECore::CompoundObject(), Plug::Flags::Default ) );
+}
+
+UDIMQuery::~UDIMQuery()
+{
+}
+
+GafferScene::ScenePlug *UDIMQuery::scenePlug()
+{
+	return getChild<ScenePlug>( g_firstPlugIndex );
+}
+
+const GafferScene::ScenePlug *UDIMQuery::scenePlug() const
+{
+	return getChild<ScenePlug>( g_firstPlugIndex );
+}
+
+Gaffer::StringPlug *UDIMQuery::uvNamePlug()
+{
+	return getChild<StringPlug>( g_firstPlugIndex + 1 );
+}
+
+const Gaffer::StringPlug *UDIMQuery::uvNamePlug() const
+{
+	return getChild<StringPlug>( g_firstPlugIndex + 1 );
+}
+
+Gaffer::StringPlug *UDIMQuery::extraAttributesPlug()
+{
+	return getChild<StringPlug>( g_firstPlugIndex + 2 );
+}
+
+const Gaffer::StringPlug *UDIMQuery::extraAttributesPlug() const
+{
+	return getChild<StringPlug>( g_firstPlugIndex + 2 );
+}
+
+GafferScene::FilterPlug *UDIMQuery::filterPlug()
+{
+	return getChild<FilterPlug>( g_firstPlugIndex + 3 );
+}
+
+const GafferScene::FilterPlug *UDIMQuery::filterPlug() const
+{
+	return getChild<FilterPlug>( g_firstPlugIndex + 3 );
+}
+
+Gaffer::CompoundObjectPlug *UDIMQuery::udimInformationPlug()
+{
+	return getChild<CompoundObjectPlug>( g_firstPlugIndex + 4 );
+}
+
+const Gaffer::CompoundObjectPlug *UDIMQuery::udimInformationPlug() const
+{
+	return getChild<CompoundObjectPlug>( g_firstPlugIndex + 4 );
+}
+
+void UDIMQuery::affects( const Plug *input, AffectedPlugsContainer &outputs ) const
+{
+	ComputeNode::affects( input, outputs );
+
+	if(
+		input == uvNamePlug() ||
+		input == extraAttributesPlug() ||
+		input == filterPlug() ||
+		input == scenePlug()->objectPlug() ||
+		input == scenePlug()->attributesPlug()
+	)
+	{
+		outputs.push_back( udimInformationPlug() );
+	}
+}
+
+namespace
+{
+
+struct InfoHashAccumulator
+{
+	bool operator()( const GafferScene::ScenePlug *scene, const GafferScene::ScenePlug::ScenePath &path )
+	{
+		IECore::MurmurHash locationHash;
+		scene->objectPlug()->hash( locationHash );
+		scene->attributesPlug()->hash( locationHash );
+		locationHash.append( scene->objectHash( path ) );
+		std::string pathStr;
+		ScenePlug::pathToString( path, pathStr );
+		m_hashes.push_back( std::pair<std::string, IECore::MurmurHash>( pathStr, locationHash ) );
+		return true;
+	}
+
+	void appendHash( IECore::MurmurHash &h )
+	{
+		std::sort( m_hashes.begin(), m_hashes.end() );
+		for( const auto &i : m_hashes )
+		{
+			h.append( i.second );
```

Comment:
Do we need to include `i.first` here too? If so, can you add to the test cases to expose the bug?